packages/core/src/StateMachine.ts

```diff
@@ -240,32 +240,42 @@ export class StateMachine<
     return transitionNode(this.root, state.value, state, _event) || [];
   }
 
-  public get first(): State<TContext, TEvent> {
-    const pseudoinitial = this.resolveState(
+  private get preInitialState(): State<TContext, TEvent> {
```

Comment:
q: how do we define a "pre initial state"? could you add a comment above this method that would explain this?

Reply:
Added a comment. This is the initial state _before_ evaluating any microsteps. The `preInitialState` may differ from the `initialState` e.g. if the `preInitialState` has an eventless transition to a different state.